File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
// 이분탐색

function solution(people, limit) {
  let answer = 0;

  people.sort((a, b) => a - b);

  let start = 0;
  let last = people.length - 1;

  while (start < last) {
    const sumValue = people[start] + people[last];
    // 더한 값이 limit보다 크면 작은 값 하나만 탈출
    if (sumValue > limit) {
      last--;
      // 그 외에는 양 끝값 탈출
    } else {
      start++;
      last--;
    }
    answer++;
  }

  if (start === last) answer++;

  return answer;
}

File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    
    result = deque(sorted(people))
    
    while len(result):
        # 사람이 한 명이라면
        if len(result) == 1:
            answer += 1
            break
        
        # 사람이 둘 이상이라면
        if result[0] + result[-1] > limit:
            answer += 1
            result.pop()
        else:
            answer += 1
            result.popleft()
            result.pop()
    
    return answer